### ISSUE TITLE
Uncaught Error: Module version mismatch. Expected 41, got 17.

### ISSUE THREAD

**Issue by szhorvat** [Galadirith/markdown-preview-plus#22]
The following error comes up whenever trying to turn on MarkDown Preview with math since Atom version 0.177.  Atom 0.176 worked fine earlier today.

**Atom Version**: 0.177.0
**System**: Mac OS X 10.10.2
**Thrown From**: [markdown-preview-plus](https://github.com/Galadirith/markdown-preview-plus) package, v1.1.0

### Stack Trace

Uncaught Error: Module version mismatch. Expected 41, got 17.

```
At /Applications/Atom.app/Contents/Resources/atom/common/lib/asar.js:422

Error: Module version mismatch. Expected 41, got 17.
  at Error (native)
  at Object.module.(anonymous function) [as .node] (/Applications/Atom.app/Contents/Resources/atom/common/lib/asar.js:422:20)
  at Module.load (/Applications/Atom.app/Contents/Resources/app/node_modules/coffee-script/lib/coffee-script/register.js:45:36)
  at Function.Module._load (module.js:325:12)
  at Module.require (module.js:380:17)
  at require (module.js:399:17)
  at Object.<anonymous> (/Users/szhorvat/.atom/packages/markdown-preview-plus/node_modules/pathwatcher/lib/main.js:6:13)
  at Object.<anonymous> (/Users/szhorvat/.atom/packages/markdown-preview-plus/node_modules/pathwatcher/lib/main.js:207:4)
  at Module._compile (module.js:475:26)
  at Object.loadFile [as .js] (/Applications/Atom.app/Contents/Resources/app/src/6to5.js:140:21)
  at Module.load (/Applications/Atom.app/Contents/Resources/app/node_modules/coffee-script/lib/coffee-script/register.js:45:36)
  at Function.Module._load (module.js:325:12)
  at Module.require (module.js:380:17)
  at require (module.js:399:17)
  at Object.<anonymous> (/Users/szhorvat/.atom/packages/markdown-preview-plus/lib/markdown-preview-view.coffee:8:10)
  at Object.<anonymous> (/Users/szhorvat/.atom/packages/markdown-preview-plus/lib/markdown-preview-view.coffee:1:1)
  at Module._compile (module.js:475:26)
  at Object.requireCoffeeScript [as .coffee] (/Applications/Atom.app/Contents/Resources/app/src/coffee-cache.js:74:19)
  at Module.load (/Applications/Atom.app/Contents/Resources/app/node_modules/coffee-script/lib/coffee-script/register.js:45:36)
  at Function.Module._load (module.js:325:12)
  at Module.require (module.js:380:17)
  at require (module.js:399:17)
  at isMarkdownPreviewView (/Users/szhorvat/.atom/packages/markdown-preview-plus/lib/main.coffee:13:26)
  at Object.module.exports.toggle (/Users/szhorvat/.atom/packages/markdown-preview-plus/lib/main.coffee:82:8)
  at atom-workspace.atom.commands.add.markdown-preview-plus:toggle (/Users/szhorvat/.atom/packages/markdown-preview-plus/lib/main.coffee:45:10)
  at CommandRegistry.module.exports.CommandRegistry.handleCommandEvent (/Applications/Atom.app/Contents/Resources/app/src/command-registry.js:243:29)
  at /Applications/Atom.app/Contents/Resources/app/src/command-registry.js:3:61
  at CommandPaletteView.module.exports.CommandPaletteView.confirmed (/Applications/Atom.app/Contents/Resources/app/node_modules/command-palette/lib/command-palette-view.js:120:32)
  at CommandPaletteView.module.exports.SelectListView.confirmSelection (/Users/szhorvat/.atom/packages/markdown-preview-plus/node_modules/atom-space-pen-views/lib/select-list-view.js:335:21)
  at space-pen-div.atom.commands.add.core:confirm (/Users/szhorvat/.atom/packages/markdown-preview-plus/node_modules/atom-space-pen-views/lib/select-list-view.js:109:19)
  at CommandRegistry.module.exports.CommandRegistry.handleCommandEvent (/Applications/Atom.app/Contents/Resources/app/src/command-registry.js:243:29)
  at /Applications/Atom.app/Contents/Resources/app/src/command-registry.js:3:61
  at KeymapManager.module.exports.KeymapManager.dispatchCommandEvent (/Applications/Atom.app/Contents/Resources/app/node_modules/atom-keymap/lib/keymap-manager.js:558:16)
  at KeymapManager.module.exports.KeymapManager.handleKeyboardEvent (/Applications/Atom.app/Contents/Resources/app/node_modules/atom-keymap/lib/keymap-manager.js:396:22)
  at HTMLDocument.module.exports.WindowEventHandler.onKeydown (/Applications/Atom.app/Contents/Resources/app/src/window-event-handler.js:170:20)

```

### Commands

```
     -0:05.5 tree-view:toggle (atom-text-editor.editor)
     -0:02.0 command-palette:toggle (ul.list-inline.tab-bar.inset-panel)
     -0:00.7 core:confirm (atom-text-editor.editor.mini)
     -0:00.7 markdown-preview-plus:toggle (ul.list-inline.tab-bar.inset-panel)
```

### Config

```json
{
  "core": {
    "disabledPackages": [
      "markdown-preview",
      "vim-mode",
      "recent-files"
    ],
    "themes": [
      "atom-light-ui",
      "atom-light-syntax"
    ]
  },
  "markdown-preview-plus": {
    "enableLatexRenderingByDefault": true
  }
}
```

### Installed Packages

```coffee
# User
language-matlab, v0.1.0
markdown-preview-plus, v1.1.0
mathjax-wrapper, v0.1.0
minimap, v4.0.2
open-recent, v2.2.0

# Dev
No dev packages
```

/cc @atom/core

**Comment by Galadirith** [comment]
@szhorvat Thanks for the report, definitely looks like a show stopper (its happing on my installations to) and I'm digging into it right now to find the cause of the error. [markdown-preview](https://github.com/atom/markdown-preview) is working fine, so its definitly a change i've made.

Thanks again @szhorvat, i'll update the issue as soon as I'm can with more information.

**Comment by Galadirith** [comment]
@szhorvat thanks for your patience. I've push a new version that should fix this (It looks like it was the [pathwatcher](https://github.com/atom/node-pathwatcher) dependency that had to be updated).

**Comment by szhorvat** [comment]
Confirming that 1.2.0 works now.

**Comment by jjwon** [comment]
I'm on Atom 0.181.0, and 1.2.0, but I'm still getting this error when I try to use <code>ctrl-shift-m</code>.  After the initial error appears, if I press it again, it gives me an error saying "Module did not self-register" from <code>/Applications/Atom.app/Contents/Resources/atom/common/lib/asar.js:422</code>.  I'm assuming the subsequent error comes from the initial one?

**Comment by Galadirith** [comment]
@jjwon I so sorry I haven't been able to respond in nearly a week and thanks for flagging this.

Its likely the package's dependencies are mismatched with Atoms cached dependencies (which is what I believe the problem was last time). I'll be pushing out an update tomorrow which will hopefully fix that for you.

In the mean time, I have checked on two machines running Atom 0.182.0 and 1.2.0 and they both work without errors. They were fresh installations so you might like to try uninstalling markdown-preview-plus and reinstalling in the mean time.

Please let me know if you have any other problems and I'll update the issue when I've rolled out a new version. Thanks again @jjwon :D

**Comment by jjwon** [comment]
No problem.  After encountering this problem, I accidentally deleted all my Atom packages, but when I reinstalled them, everything worked fine.  I still don't know what caused the original problem (it started when I was updating Atom), but at least for me it is now fixed.

When I had the issue, uninstalling and reinstalling mpp did not solve the problem, though.

Thanks for all your work by the way!  Love this add-on.
